app/controllers/admin/cases_controller.rb

```diff
@@ -30,55 +34,118 @@ def index
   end
 
   def new
-    case_creator = CTS::Cases::Create.new(Rails.logger, case_model: Case::Base)
+    if params[:correspondence_type].present?
+      @correspondence_type = params[:correspondence_type]
+      self.send("prepare_new_#{@correspondence_type}".to_sym)
+    else
+      prepare_select_type
+    end
+  end
+
+  def prepare_new_foi
```

Comment:
Should `prepare_new_foi` and `prepare_new_sar` be private?

Reply:
probably!